OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java

```diff
@@ -2264,7 +2288,10 @@ private int[] findActiveIndex(RouteSegmentResult prevSegm, RouteSegmentResult cu
 		if (rs == null) {
 			return pair;
 		}
-		int[] directions = getUniqTurnTypes(turnLanes);
+		int[] directions = getUniqDirections(turnLanes);
+		if (rs.roadsOnLeft + rs.roadsOnRight >= directions.length) {
```

Comment:
This is too broad condition which not correct for many correctly mapped intersections, turn:lanes could have "TL|C|TR" even for 2 way intersections cause TL could be further away, so replacing it with syntetic is not a good thing

Reply:
It is not this case. Your case when we have less roads than a turn lanes on intersection. Such situation is working fine in our code.
Function findActiveIndex need for range of active lanes.